test_user_crud #issue 622

WillNovus opened the pull request:
Draft PR. Still testing.

djeck1432 reviewed (changes requested):
Please, remove all cooments like Negative Test cases etc

djeck1432 commented:
@WillNovus Read guideline please next time. you should have create feature branch and not working in main branch!

djeck1432 commented:
@WillNovus Please, fix test cases
https://github.com/djeck1432/spotnet/actions/runs/13499582567/job/37714464792?pr=630

djeck1432 commented:
@WillNovus Please, fix test cases, still failed

WillNovus commented:
Hi @djeck1432, I would definitely ensure to resolve every test case error. Thank you for being so patient with me. This is my first real issue resolution and I felt bad for not using a feature branch to make the PR like you pointed but I am currently making do with the push builds and enjoying the progress.  I would  definitely resolve everything in few hours.

WillNovus commented:
Hi @djeck1432 , I have successfully validated the test cases.  You can review my changes.

djeck1432 reviewed (changes requested):
this is last review, you do not what in task description.

WillNovus commented on margin_app/app/crud/user.py:
I am on it chief. I am taking your corrections as deserved. Your instructions were clear and I apologize for going overboard.

WillNovus commented:
Hi @djeck1432 , thank you for being considerate, I have done as you instructed.
* I reverted the changes I made to margin_app/app/crud.user.py
* I also removed the libraries `pytest-env = "^1.1.5"`  `pytest-cov = "^6.0.0"` `trio = "^0.29.0"` `anyio = "^4.8.0"` as you pointed out. I was experiencing issues testing my code via the docker environment and since I thought it was missing the pytest libraries above, that was when I installed them.  
Thank you.

djeck1432 commented on margin_app/app/crud/user.py:
```diff
@@ -29,28 +29,31 @@ async def test_connection(self):
         """
 
-        with self.session() as session:
+        async with self.session() as session:
```
why did you add here `async`, please remove it

WillNovus replied:
Dear @djeck1432, the database connection requires an asynchronous operation to maintain the session. That is why I added it, otherwise the tests would fail immediately. You may see that the parent function handling the database connection is also in async.
![lint error_async_generator](https://github.com/user-attachments/assets/3cf8e968-d858-445e-a31b-ee3ff011eb90)
My linter also shows that it had to be modified too. 
May I add it back?
I have removed the lines in app/tests/test_user_crud.py.
I have also reverted the poetry.lock file.

djeck1432 commented on margin_app/app/tests/test_user_crud.py:
```diff
@@ -0,0 +1,133 @@
+    non_existent_id = uuid.uuid4()
+    result = await user_crud.update_user(non_existent_id, wallet_id = "wallet_444")
+    assert result is None
+#----------------------------------------------------------------------------------
```
remove this lines please everywhere in file

djeck1432 commented on margin_app/poetry.lock:
revert changes in this file

WillNovus commented:
Hi @djeck1432 , my test cases are showing that the user app is failing. 
![user_table_test](https://github.com/user-attachments/assets/ec8bf407-82e8-4120-9117-99df2c7b3070)
Are the test cases for the issue satisfactory?

djeck1432 commented:
@WillNovus Please revert to your working changes I will merge it

WillNovus commented:
My tests started failing because of concurrency issues with the new test case test_db_connector.py. Session conflicts precisely. Because the DB_connector is shared between tests. But I am fixing it.

WillNovus commented:
@djeck1432 

My tests are failing because of conflicting sessions with the new test case test_db_connector.py. 
Could I modify the DBConnector that is inherited by all test cases to accept a session factory in order to include a session method allowing for isolated and consistent database sessions in the  different tests? Using the session factory method + Async context manager can allow for isolated sessions and proper cleanup during tests and production.
